addons/account/wizard/accrued_orders.py

```diff
@@ -251,6 +252,8 @@ def create_entries(self):
 
         if self.reversal_date <= self.date:
             raise UserError(_('Reversal date must be posterior to date.'))
+        if len(self.currency_id) != 1:
```

Comment:
```suggestion
        if len({order.currency_id or order.company_id.currency_id for order in self}) != 1:
```
Otherwise, if you mix one order having a foreign currency and one without any currency, the error won't be raised